src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -87,34 +96,34 @@ public static bool IsRequestKeyValid(string value)
             {
                 return false;
             }
-            try
-            {
-                byte[] data = Convert.FromBase64String(value);
-                return data.Length == 16;
-            }
-            catch (Exception)
-            {
-                return false;
-            }
+
+            Span<byte> temp = stackalloc byte[16];
+            var success = Convert.TryFromBase64String(value, temp, out var written);
+            return success && written == 16;
         }
 
         public static string CreateResponseKey(string requestKey)
         {
             // "The value of this header field is constructed by concatenating /key/, defined above in step 4
-            // in Section 4.2.2, with the string "258EAFA5- E914-47DA-95CA-C5AB0DC85B11", taking the SHA-1 hash of
+            // in Section 4.2.2, with the string "258EAFA5-E914-47DA-95CA-C5AB0DC85B11", taking the SHA-1 hash of
             // this concatenated value to obtain a 20-byte value and base64-encoding"
             // https://tools.ietf.org/html/rfc6455#section-4.2.2
 
-            if (requestKey == null)
-            {
-                throw new ArgumentNullException(nameof(requestKey));
-            }
-
             using (var algorithm = SHA1.Create())
             {
-                string merged = requestKey + "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
-                byte[] mergedBytes = Encoding.UTF8.GetBytes(merged);
-                byte[] hashedBytes = algorithm.ComputeHash(mergedBytes);
+                // requestKey is already verified to be small (24 bytes) by 'IsRequestKeyValid()' and everything is 1:1 mapping to UTF8 bytes
+                // so this can be hardcoded to 60 bytes for the requestKey + static websocket string
+                Span<byte> mergedBytes = stackalloc byte[60];
+                Encoding.UTF8.GetBytes(requestKey, mergedBytes);
+                _encodedWebSocketKey.CopyTo(mergedBytes.Slice(24));
```

Comment:
What if this is called with a requestKey that isn't 24 bytes? Should we check the return value of GetBytes?

Reply:
"requestKey is already verified to be small (24 bytes) by 'IsRequestKeyValid()' and everything is 1:1 mapping to UTF8 bytes"